### PR TITLE
Feat/645 check checker ignore bool

### PR DESCRIPTION
**Pull request by cylewitruk**
Fixes:  #645

### CODE REVIEW

File: components/clarity-repl/src/analysis/check_checker/mod.rs [hunk 1]
```diff
@@ -371,7 +371,9 @@ impl<'a> ASTVisitor<'a> for CheckChecker<'a, '_> {
         // Upon entering a public function, all parameters are tainted
         if let Some(params) = parameters {
             for param in params {
-                self.add_taint_source(Node::Symbol(param.name), param.decl_span);
+                if param.type_expr.match_atom() != Some(&ClarityName::from("bool")) {
+                    self.add_taint_source(Node::Symbol(param.name), param.decl_span)
+                }
```
Comment (obycode) [inline components/clarity-repl/src/analysis/check_checker/mod.rs:376]: Let's standardize this for more flexibility in the future. Maybe something like this:

```rust
                match TypeSignature::parse_type_repr(param.type_expr, &mut ()) {
                    Ok(TypeSignature::BoolType) => (),
                    Ok(_) => self.add_taint_source(Node::Symbol(param.name), param.decl_span),
                    _ => {
                        // This error will have already been reported by the type-checker. 
                        // Just stop traversing.
                        return false;
                    }
                }
```

File: components/clarity-repl/src/analysis/check_checker/mod.rs [hunk 2]
```diff
@@ -3023,6 +3025,31 @@ mod tests {
...
 
+    #[test]
+    fn allow_unchecked_bool_in_public_function() {
+        let mut settings = SessionSettings::default();
+        settings.repl_settings.analysis.passes = vec![Pass::CheckChecker];
+        let mut session = Session::new(settings);
+        let snippet = "
+(define-public (allow_tainted (amount bool))
+    (ok amount)
+)
```
Comment (obycode) [inline components/clarity-repl/src/analysis/check_checker/mod.rs:3036]: This contract would not trigger a check-checker warning now. Returning tainted data from a public function is allowed. How about using the tainted data to set a data var instead:

```clarity
(define-data-var myvar bool false)
(define-public (tainted-var-set (b bool))
    (ok (var-set myvar b))
)
```

**Review by obycode — COMMENTED** [review]
Thanks for this PR! I have a couple suggestions to improve it.

**Comment by hugocaillard** [comment]
Took the liberty to change the description so that it's linked to the issues (https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

File: components/clarity-repl/src/analysis/check_checker/mod.rs (hunk 2 above)
Comment (cylewitruk) [reply components/clarity-repl/src/analysis/check_checker/mod.rs]: Ah didn't know that, I updated to use your example.. and learned something 😅

**Review by obycode — COMMENTED** [review]
LGTM! Thanks for the updates. The new commits came from a different user and only 1 of the users has signed the CLA. Can you fix that by either signing for both or re-committing with the other user and force pushing?

**Comment by obycode** [comment]
Also, please run `cargo fmt` (see [failing check](https://github.com/hirosystems/clarinet/actions/runs/3427688613/jobs/5713338122)).

**Comment by cylewitruk** [comment]
> LGTM! Thanks for the updates. The new commits came from a different user and only 1 of the users has signed the CLA. Can you fix that by either signing for both or re-committing with the other user and force pushing?

Well that didn't turn out so pretty 😅

**Comment by obycode** [comment]
> Well that didn't turn out so pretty 😅

Yeah, I think something went wrong there 😄

**Comment by cylewitruk** [comment]
> > Well that didn't turn out so pretty 😅
> 
> Yeah, I think something went wrong there 😄

Will re-do with a new PR.. easiest since it's so small changes..

**Pull request closed without merging**